### PR TITLE
python3Packages.flask-mysqldb: init at 2.0.0

### PR DESCRIPTION
**Pull request by NetaliDev**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/flask-mysqldb/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+, buildPythonPackage
+, fetchFromGitHub
+, flask
+, mysqlclient
+}:
+
+buildPythonPackage rec {
+  pname = "flask-mysqldb";
+  version = "1.0.1";
+  format = "setuptools";
```
Comment (fabaff) [inline pkgs/development/python-modules/flask-mysqldb/default.nix:11]: Upstream says [not explicitly](https://github.com/alexferl/flask-mysqldb/blob/master/setup.py#L35C1-L35C49) that the module is not working on Python < 3.7. Still, let's stay away from unsupported Python releases for new packages.

```suggestion
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/flask-mysqldb/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [
+    flask
+    mysqlclient
+  ];
+
+  pythonImportsCheck = [
+    "flask_mysqldb"
+  ];
+
```
Comment (fabaff) [inline pkgs/development/python-modules/flask-mysqldb/default.nix:28]: Module has no unittests.

```suggestion

  # Module has no tests
  doCheck = false;
```

File: pkgs/development/python-modules/flask-mysqldb/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  meta = with lib; {
+    description = "MySQL connection support for Flask";
+    homepage = "https://github.com/alexferl/flask-mysqldb";
```
Comment (fabaff) [inline pkgs/development/python-modules/flask-mysqldb/default.nix:31]: Tools (e.g., `nix-update`) can pick up the changelog entry during update and add the link to the changelog to the commit message.

```suggestion
    homepage = "https://github.com/alexferl/flask-mysqldb";
    changelog = "https://github.com/alexferl/flask-mysqldb/releases/tag/v${version}";
```

File: pkgs/development/python-modules/flask-mysqldb/default.nix (hunk 1 above)
Comment (NetaliDev) [reply pkgs/development/python-modules/flask-mysqldb/default.nix:12]: Fixed.

File: pkgs/development/python-modules/flask-mysqldb/default.nix (hunk 2 above)
Comment (NetaliDev) [reply pkgs/development/python-modules/flask-mysqldb/default.nix:31]: Fixed.

File: pkgs/development/python-modules/flask-mysqldb/default.nix (hunk 3 above)
Comment (NetaliDev) [reply pkgs/development/python-modules/flask-mysqldb/default.nix:37]: Fixed.

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 258288` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.flask-mysqldb</li>
    <li>python310Packages.flask-mysqldb.dist</li>
    <li>python311Packages.flask-mysqldb</li>
    <li>python311Packages.flask-mysqldb.dist</li>
  </ul>
</details>

**Review by mbalatsko — APPROVED** [review]
LGTM

File: pkgs/development/python-modules/flask-mysqldb/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+, pythonOlder
+, fetchFromGitHub
+, flask
+, mysqlclient
+}:
+
+buildPythonPackage rec {
+  pname = "flask-mysqldb";
+  version = "1.0.1";
+  format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/flask-mysqldb/default.nix:12]:
```suggestion
  pyproject = true;
```
FYI: A new parameter, pyproject, has recently been introduced.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

**Review by natsukium — APPROVED** [review]
LGTM

File: pkgs/development/python-modules/flask-mysqldb/default.nix (hunk 4 above)
Comment (NetaliDev) [reply pkgs/development/python-modules/flask-mysqldb/default.nix]: Thanks for the hint! I use the new parameter now.

**Comment by NetaliDev** [comment]
It seems like version 2.0.0 released yesterday --> updated PR to 2.0.0

**Comment by NetaliDev** [comment]
Hmm, I can't get version 2.0.0 to work locally. Draft until I figure out the reason,

**Comment by NetaliDev** [comment]
Got it, it seems like the usage of the module changed slightly.

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 258288` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.flask-mysqldb</li>
    <li>python310Packages.flask-mysqldb.dist</li>
    <li>python311Packages.flask-mysqldb</li>
    <li>python311Packages.flask-mysqldb.dist</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**